### PR TITLE
Create a new user

### PR DESCRIPTION
**Pull request by LornaHarwood**
Created a Sign Up feature.
Link within the Navbar.
As a first pass I have enabled a user creation with just a name.

### CODE REVIEW

**Comment by LornaHarwood** [comment]
[As a person who uses the application, I want to be able to create a user](https://trello.com/c/XgLLwFnv/30-as-a-person-who-uses-the-application-i-want-to-be-able-to-create-a-user)

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<h1>Create User</h1>
+
+<%= form_with model: @user, local: true do |form| %>
+
+  <p>
+    <%= form.label :name %><br>
+    <%= form.text_field :name %>
+  </p>
```
Comment (erbridge) [inline app/views/users/new.html.erb:8]:
```suggestion
  <h2>
    <%= form.label :name %>
  </h2>
  <p>
    <%= form.text_field :name %>
  </p>
```

Semantic HTML, remember :).

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -13,6 +13,7 @@
         <ul>
           <li>
             <a href="/pages/new">Create Page</a>
+            <a href="/users/new">Sign Up</a>
```
Comment (erbridge) [inline app/views/layouts/application.html.erb:16]: Shouldn't this be a separate `<li>`?

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<h1>Create User</h1>
+
+<%= form_with model: @user, class: "page-form", local: true do |form| %>
```
Comment (erbridge) [inline app/views/users/new.html.erb:3]:
```suggestion
<%= form_with model: @user, class: "user-form", local: true do |form| %>
```

If you want them to have the same styles, either make the selector in your CSS more generic, or apply the styles to more classes.

File: spec/features/user_can_create_other_users_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require "rails_helper"
```
Comment (erbridge) [inline spec/features/user_can_create_other_users_spec.rb:1]: This file's name isn't very clear about what we're testing here. It sounds like I should be able to create user accounts on behalf of someone else.

File: spec/features/user_can_create_other_users_spec.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      visit pages_path
+
+      expect(page).to have_content("Sign Up")
+
+      click_link "Sign Up"
+
+      expect(page).to have_content("Create User")
+      expect(page).to have_content("Name")
+
+      fill_in "Name", with: "Name"
```
Comment (erbridge) [inline spec/features/user_can_create_other_users_spec.rb:15]:
```suggestion
      fill_in "Name", with: "Test user"
```

Better to test with something that would never be part of the defaults or the markup.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (LornaHarwood) [reply app/views/layouts/application.html.erb:16]: Ah, yes! thanks for spotting

File: spec/features/user_can_create_other_users_spec.rb (hunk 4 above)
Comment (LornaHarwood) [reply spec/features/user_can_create_other_users_spec.rb:1]: I struggled naming this file. Do you have any suggestions at all?

Comment (erbridge) [reply spec/features/user_can_create_other_users_spec.rb:1]: `user_can_create_a_user_spec`?

Comment (erbridge) [reply spec/features/user_can_create_other_users_spec.rb:1]: Or `visitor_can_create_user_spec`. No harm in being more specific in what we mean by "user" in our story.

File: app/assets/stylesheets/pages.scss [hunk 6]
```diff
@@ -139,7 +139,7 @@ main {
   }
 }
 
-.page-form {
+.page-form, .user-form {
```
Comment (erbridge) [inline app/assets/stylesheets/pages.scss:142]: (Prettier - it will save you pain later)

**Pull request merged**